keps/sig-auth/2862-fine-grained-kubelet-authz/README.md

```diff
@@ -0,0 +1,1013 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+We propose a change to the Kubelet API authorization to provide more
+fine-grained control. Currently, the API uses a coarse authorization scheme,
+where actions like reading health status and the ability to exec into a pod
+require the same permission.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The Kubelet API on the authenticated(10250) port currently uses a very coarse
+scheme for authorizing requests. For example, reading `/healthz` and calling
+`/exec/…` (i.e. execute arbitrary code) require the same `proxy` subresource.
+So, if an application needs to say `list` `pods` on the node, as many monitoring
+and logging agents do, they must be granted the `proxy` subresource in RBAC.
+Doing so grants many other powerful permissions to these agents which could be
+exploited by an attacker to escalate privilege.
```

Comment:
Might be worth adding the historical context: healthz and pods used to be availabel on the read-only port. With the removal of the read only port, it's raised the priority of this issue.

Reply:
Done. Thanks!